### PR TITLE
feat: Support for slots in SViews

### PR DESCRIPTION
**Pull request by soeren-schmaljohann-2denker**

### CODE REVIEW

**Comment by ABausG** [comment]
The goal for this is to be explicitly **not** a breaking change as we need to allow a phase where the app still supports both format.
The strategy should follow these steps:

- Keep the current constructor arguments but assign them to nullable private fields, mark them as deprecated and adjust the current variables to be getters that use the probate implementation of available or construct them through the new slot variant
- move some of the required keywords to be represented as assertions such that there is no breaking change but still a sort of safety for creating it

Also I remember that @bocasti mentioned that the data in principle is still the same so possibly we don't even need a new data type for the slots but can instead still use the old data? If that's the case we might even keep the current structure (maybe renaming some variables and deprecating old names) and only having to adjust the json constructors

**Comment by soeren-schmaljohann-2denker** [comment]
@ABausG I refactored it to no longer be a breaking change

**Comment by bocasti** [comment]
> The goal for this is to be explicitly **not** a breaking change as we need to allow a phase where the app still supports both format. The strategy should follow these steps:
> 
> * Keep the current constructor arguments but assign them to nullable private fields, mark them as deprecated and adjust the current variables to be getters that use the probate implementation of available or construct them through the new slot variant
> * move some of the required keywords to be represented as assertions such that there is no breaking change but still a sort of safety for creating it
> 
> Also I remember that @bocasti mentioned that the data in principle is still the same so possibly we don't even need a new data type for the slots but can instead still use the old data? If that's the case we might even keep the current structure (maybe renaming some variables and deprecating old names) and only having to adjust the json constructors



> The goal for this is to be explicitly **not** a breaking change as we need to allow a phase where the app still supports both format. The strategy should follow these steps:
> 
> * Keep the current constructor arguments but assign them to nullable private fields, mark them as deprecated and adjust the current variables to be getters that use the probate implementation of available or construct them through the new slot variant
> * move some of the required keywords to be represented as assertions such that there is no breaking change but still a sort of safety for creating it
> 
> Also I remember that @bocasti mentioned that the data in principle is still the same so possibly we don't even need a new data type for the slots but can instead still use the old data? If that's the case we might even keep the current structure (maybe renaming some variables and deprecating old names) and only having to adjust the json constructors

I confirm that only the json format of the view resource should change. It should be the same content (same keys and values) but presented differently

**Review by ABausG — CHANGES_REQUESTED** [review]
See the comment. The deprectation is not yet layed out for truly allowing compatibility with both ways.

I'm attaching some responses to the ticket with data in the old/new format that should be used for the tests

**Comment by ABausG** [comment]
I just realised that we might want to just make the following:

- Adjust this to only add the slot properties and not touch the old things apart from marking them with deprectated
- DON'T release that version but instead build a version of the apptive launcher with a dependency override onto this git branch
- Do a breaking change once that is released with the changes more or less from the first Review.


Sorry for not coming up with that solution earlier. What do you think about that?

File: packages/apptive_grid_core/lib/src/model/s_view/s_view.dart [hunk 1]
```diff
@@ -34,23 +44,34 @@ class SView {
         json['_links'],
       ),
       fields: fields,
-      fieldProperties: fieldProperties,
```
[the remark this answers is not in the document]
Comment (soeren-schmaljohann-2denker) [reply packages/apptive_grid_core/lib/src/model/s_view/s_view.dart:37]: I changed it so both versions can be passed and added some tests for it

**Review by ABausG — COMMENTED** [review]
Just some last nits:

- Maybe move the migration test responses to separate files so the test file itself is a bit more readable
- I guess this is now rather a feature than a refactor. Also make sure that the spelling of the title is correct before merging and fits better as a changelog message something like: `Support slots for SViews`

**Comment by soeren-schmaljohann-2denker** [comment]
> Just some last nits:
> 
> * Maybe move the migration test responses to separate files so the test file itself is a bit more readable
> * I guess this is now rather a feature than a refactor. Also make sure that the spelling of the title is correct before merging and fits better as a changelog message something like: `Support slots for SViews`

@ABausG done

**Review by ABausG — APPROVED** [review]

**Pull request merged**